bdk_chain/src/chain_graph.rs

```diff
@@ -55,13 +56,27 @@ impl<P: ChainPosition> ChainGraph<P> {
     /// Inserts a transaction into the inner [`ChainGraph`] and optionally into the chain at
     /// `position`.
     ///
+    /// If inserting it into a chain `position` will result in a conflict, an error will be returned
+    /// instead.
+    ///
     /// **Warning**: This function modifies the internal state of the chain graph. You are
     /// responsible for persisting these changes to disk if you need to restore them.
     pub fn insert_tx(
         &mut self,
         tx: Transaction,
         position: Option<P>,
-    ) -> Result<bool, sparse_chain::InsertTxErr> {
+    ) -> Result<bool, InsertTxErr<P>> {
+        // only care about conflicts if chain position is provided
+        if position.is_some() {
```

Comment:
Why do we enforce this if position of the new tx isn't required to find conflicts?

Reply:
Sorry, I'm not sure I know exactly what you mean. Do you mind expanding?